### PR TITLE
Implement Command History

### PR DESCRIPTION
**Pull request by KamiliArsyad**
Keeps track of valid commands entered.

### CODE REVIEW

**Review by MagnificentCreature — APPROVED** [review]
LGTM

File: src/main/java/seedu/address/ui/CommandBox.java [hunk 1]
```diff
@@ -16,6 +17,8 @@ public class CommandBox extends UiPart<Region> {
     public static final String ERROR_STYLE_CLASS = "error";
     private static final String FXML = "CommandBox.fxml";
 
+    private static final CommandHistory HISTORY = new CommandHistory();
```
Comment (dickongwd) [inline src/main/java/seedu/address/ui/CommandBox.java]: Should history be all caps?

File: src/main/java/seedu/address/model/CommandHistory.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        if (commandHistory.size() != 0 && commandHistory.getLast().equals(command)) {
+            return;
+        }
+
+        // Do not add empty commands
+        if (command.equals("")) {
+            return;
+        }
+
+        removeHistoryAfterCurrentIndex();
```
Comment (dickongwd) [inline src/main/java/seedu/address/model/CommandHistory.java:49]: Not sure of the use of this function

**Review by dickongwd — APPROVED** [review]
LGTM, just some nits

**Pull request merged**